### PR TITLE
Feature/prosecco review updates

### PR DESCRIPTION
**Pull request by rkolpakov**

### CODE REVIEW

**Review by TheDZhon — CHANGES_REQUESTED** [review]
Thank you for the **tremendous** work and effort you put into making this boat moving :ship: In my view, this particular project is essential for the DAO's operational robustness, and it would be great to make it happen and rolled out after all of the back-and-forth movements :smile: 

Overall, LGTM, except for a few missing parts and improvements.

---

On top of the comments provided in the attached review:
- :warning: there should be a manager change/dismissal to mitigate potential griefing attempts by a committee.
- :bulb: please add a top-level license file (should contain the MIT terms)
- :bulb: margin and tolerable amounts terms and agreements should be ELI5-ed in the code, docs, and deploy plan, esp cool if graphic schemes were provided
- :mag: consider pinning the OZ contracts version in `package.json` strictly
- :mag: consider adding test coverage metrics
- :mag: consider adding Slither
- :mag: custom errors could have been more detailed by accepting additional arguments (e.g., if they indicate a private constant-guarded interval violation) for `Stonks` and `Order`
- :man_shrugging: please consider disambiguating Cow Swap and Cow Protocol (it looks like we should use the latter term): https://docs.cow.fi/cow-protocol/concepts/how-it-works/protocol-vs-swap

--- 

:warning: Blocking things from my side for deployment:
- Manager dismissal/removal
- Updated audit
- Updated deployment plan with gradual trade amount increase adoption

Nice to haves:
- updated spec
- all of the other fixes and suggestions (this PR + deploy plan comments)

File: contracts/AccessControl.sol [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        if (agent_ == address(0)) revert InvalidAgentAddress();
+        agent = agent_;
+    }
+
+    /**
+     * @dev Sets the manager address.
+     * @param manager_ The address of the new manager.
+     */
+    function setManager(address manager_) external onlyAgent {
+        if (manager_ == address(0)) revert InvalidManagerAddress();
```
Comment (TheDZhon) [inline contracts/AccessControl.sol:34]: let's have an ability to dismiss manager still

File: contracts/Stonks.sol [hunk 2]
```diff
@@ -79,44 +83,49 @@ contract Stonks is IStonks, AssetRecoverer {
...
 
         // Contract needs to hold at least 10 wei to cover steth shares issue
         if (balance <= MIN_POSSIBLE_BALANCE) revert MinimumPossibleBalanceNotMet();
 
         Order orderCopy = Order(Clones.clone(orderSample));
         IERC20(orderParameters.tokenFrom).safeTransfer(address(orderCopy), balance);
-        orderCopy.initialize(manager);
+        orderCopy.initialize(minBuyAmount_, manager);
+
+        emit OrderContractCreated(address(orderCopy));
```
[the remark this answers is not in the document]
Comment (rkolpakov) [reply contracts/Stonks.sol:108]: Unfortunately we can't support event interface for `cow-watch-tower` because it requires to pack specific metadata and inherit interfaces we don't use. Moreover, there is no guarantee that it will be working in a stable way and for a long time. The solution suggested by @avsetsin has been abandoned and replaced with `cow-watch-tower`. 
As a result we decided to integrate offchain order creation to EasyTrack UI and support it by ourselves.

File: contracts/AccessControl.sol (hunk 1 above)
Comment (rkolpakov) [reply contracts/AccessControl.sol:34]: Does changing manager to agent address work? Allowing to set zero address sounds a little bit scary to me.

**Review by Psirex — APPROVED** [review]
